ckanext/restricted_access/middleware.py

```diff
@@ -1,18 +1,26 @@
-import ckan.authz as authz
-import ckan.model as model
-import ckan.plugins.toolkit as toolkit
 import json
 import six
-import ckan.lib.api_token as api_token
 import logging
 
+from six.moves.urllib.parse import urlparse
 
-from ckan.common import _, config
+import ckan.authz as authz
+import ckan.model as model
+import ckan.lib.api_token as api_token
+import ckan.plugins.toolkit as toolkit
+
+from ckan.common import _, config, request
```

Comment:
You can import or use all of these from the toolkit.
eg. toolkit.config, toolkit.request
It is recommended in extensions to import from the toolkit if available instead of directly to CKAN.
https://docs.ckan.org/en/latest/extensions/best-practices.html#use-the-plugins-toolkit-instead-of-importing-ckan